packages/react-native-reanimated/__tests__/animatedProps.test.tsx

```diff
@@ -0,0 +1,84 @@
+import { fireEvent, render } from '@testing-library/react-native';
+import React from 'react';
+import { Button, TextInput, View } from 'react-native';
+import Animated, {
+  useAnimatedProps,
+  useSharedValue,
+  withTiming,
+} from 'react-native-reanimated';
+import { Circle, Svg } from 'react-native-svg';
+
+Animated.addWhitelistedNativeProps({ text: true });
+
+const AnimatedCircle = Animated.createAnimatedComponent(Circle);
+const AnimatedTextInput = Animated.createAnimatedComponent(TextInput);
+
+export default function AnimatedComponent() {
+  const r = useSharedValue(20);
+  const width = useSharedValue(20);
+
+  const handlePress = () => {
+    r.value += 10;
+    width.value += 10;
+  };
+
+  const animatedProps = useAnimatedProps(() => ({
+    r: withTiming(r.value, { duration: 100 }),
```

Comment:
Define it in the upper scope.

```suggestion
    r: withTiming(r.value, { duration: animationDuration }),
```

Reply:
yes 👍